### PR TITLE
Feat/pro 4199/oe selftest sota tools

### PR DESCRIPTION
**Pull request by patrickvacek**
I tried out a lot of different ideas beyond what I'm committing here, but I either couldn't get the ideas to work or they seemed redundant with our existing e2e infrastructure, so what's left here is what I think is the most essential and useful. The idea is that we are testing that the basic building blocks connect to each other and the bitbake system works as expected. Do let me know if you have further ideas that would make sense to include here.

### CODE REVIEW

File: scripts/run-qemu-ota [hunk 1]
```diff
@@ -2,126 +2,12 @@
 
 from argparse import ArgumentParser
 from subprocess import Popen
-from os.path import exists, join, realpath
-from os import listdir
-import random
+from os.path import exists, join
 import sys
-import socket
+import QemuCommand
```
Comment (cajun-rat) [inline scripts/run-qemu-ota]: This refactor is great, but one minor nit: can you use the [pep8](https://www.python.org/dev/peps/pep-0008/#package-and-module-names) convention for the name of the QemuCommand module, please?  e.g:

     from qemucommand import QemuCommand

**Comment by cajun-rat** [comment]
I'm still running into troubles getting this to pass. On shovel I get the following output:

~~~
2017-11-13 17:14:07 - test_deploy_help (oeqa.selftest.updater.SotaToolsTests) ... FAIL
2017-11-13 17:14:10 - test_push_help (oeqa.selftest.updater.SotaToolsTests) ... FAIL

======================================================================
FAIL: test_help (oeqa.selftest.updater.GarageSignTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/phil/pro/meta-updater/lib/oeqa/selftest/updater.py", line 48, in test_help
    self.assertTrue(os.path.isfile(p), msg = "No garage-sign found (%s)" % p)
AssertionError: False is not true : No garage-sign found (/home/phil/pro/build-qemu/tmp/work/x86_64-linux/garage-sign-native/0.2.0-29-gf6f095a-r0/sysroot-destdir/home/phil/pro/build-qemu/tmp/work/x86_64-linux/garage-sign-native/0.2.0-29-gf6f095a-r0/recipe-sysroot-native/usr/bin/garage-sign)

======================================================================
FAIL: test_deploy_help (oeqa.selftest.updater.SotaToolsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/phil/pro/meta-updater/lib/oeqa/selftest/updater.py", line 32, in test_deploy_help
    self.assertTrue(os.path.isfile(p), msg = "No garage-deploy found (%s)" % p)
AssertionError: False is not true : No garage-deploy found (/home/phil/pro/build-qemu/tmp/work/x86_64-linux/aktualizr-native/1.0+gitAUTOINC+56719e1c2d-7/sysroot-destdir/home/phil/pro/build-qemu/tmp/work/x86_64-linux/aktualizr-native/1.0+gitAUTOINC+56719e1c2d-7/recipe-sysroot-native/usr/bin/garage-deploy)

======================================================================
FAIL: test_push_help (oeqa.selftest.updater.SotaToolsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/phil/pro/meta-updater/lib/oeqa/selftest/updater.py", line 25, in test_push_help
    self.assertTrue(os.path.isfile(p), msg = "No garage-push found (%s)" % p)
AssertionError: False is not true : No garage-push found (/home/phil/pro/build-qemu/tmp/work/x86_64-linux/aktualizr-native/1.0+gitAUTOINC+56719e1c2d-7/sysroot-destdir/home/phil/pro/build-qemu/tmp/work/x86_64-linux/aktualizr-native/1.0+gitAUTOINC+56719e1c2d-7/recipe-sysroot-native/usr/bin/garage-push)

----------------------------------------------------------------------
Ran 7 tests in 524.884s

FAILED (failures=3)
~~~

and on my machine I get 

~~~
017-11-13 17:02:24 - test_push_help (oeqa.selftest.updater.SotaToolsTests) ... ok

======================================================================
FAIL: test_add_package (oeqa.selftest.updater.GeneralTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/phil/pro/poky/../meta-updater/lib/oeqa/selftest/updater.py", line 76, in test_add_package
    self.assertEqual(result.output, 'man: /usr/bin/man')
AssertionError: 'WARNING: Duplicate inclusion for /home/ph[1352 chars]/man' != 'man: /usr/bin/man'
Diff is 1408 characters long. Set self.maxDiff to None to see it.

----------------------------------------------------------------------
Ran 7 tests in 285.219s

FAILED (failures=1)
~~~

**Comment by patrickvacek** [comment]
Fixed the PEP 8 issue, good call. (However, 'run-qemu-ota' breaks other Python standards due to the dashes, which is the entire reason I did the refactoring.)

Haven't figured out the bigger problems. What does `find tmp -type f -name garage-sign` and the same for garage-push and garage-deploy yield on shovel?

**Comment by patrickvacek** [comment]
I may have a fix for your local `test_add_package` build. Let me know if that works.

And I may need to get set up on shovel to diagnose the rest.

**Comment by patrickvacek** [comment]
Okay, the `test_add_package` issue is caused by duplicate lines in your `bblayers.conf`. You need to remove them. I think I have an idea what causes that problem and will look into it after I get the rest of this working.

**Comment by cajun-rat** [comment]
It turns our that adding `INHERIT += "rm_work"` to local.conf (or site.conf) is a Bad Idea. Without that line the tests pass.

**Comment by patrickvacek** [comment]
In case you ever need `rm_work` again for some ungodly reason, you can use `RM_WORK_EXCLUDE_ITEMS` in `local.conf` (or presumably even directly in the oe-selftest `updater.py`) to keep whatever is necessary for the tests.

**Comment by cajun-rat** [comment]
Cool, thanks

**Pull request merged**